ffcx/analysis.py

```diff
@@ -246,7 +248,7 @@ def _analyze_form(
 
 
 def _has_custom_integrals(
-    o: ufl.integral.Integral | ufl.classes.Form | list | tuple,
+    o: typing.Union[ufl.integral.Integral, ufl.classes.Form, list, tuple],
```

Comment:
How was this not causing an error previously? Or is 3.9 not running on CI so we just weren't seeing the error?

Is it worth considering bumping the minimun version to 3.10 six months early so we can use `|`?

Reply:
I think it depends on what functions are typed. Mypy ignores that are typed (if they are within an untyped function signature).